infrastructure/resource_group_level/modules/vnet.bicep

```diff
@@ -0,0 +1,218 @@
+
+param appServicePlanName string
+
+param routeTableId string
+param networkSecurityGroupId string
+// TODO:DTFS-6422 Note that none of these parameters/values seem to appear in the old IaC scripts
+param demoGatewayPrefixCidr string
+param demoPrivateEndpointsPrefixCidr string
+param peeringAddressSpace string
+param storageLocations array
+
+var vnetName = 'tfs-${appServicePlanName}-vnet'
+var maintenanceVmSubnetName = '${resourceGroup().name}-vm'
+
+var appServicePlanEgressSubnetName = '${appServicePlanName}-app-service-plan-egress'
+var gatewaySubnetName = '${environment}-gateway'
+var privateEndpointsSubnetName = '${environment}-private-endpoints'
+
+// TODO:DTFS-6422 - I don't think this is explicitly named in the scripts. We should come up with a better name.
+var vnetPeeringName = 'tfs-dev-vnet_vnet-ukef-uks'
+
+resource vnet 'Microsoft.Network/virtualNetworks@2022-11-01' = {
+  name: vnetName
+  location: location
+  tags: {
+    Environment: 'Preproduction'
+  }
+  properties: {
+    addressSpace: {
+      addressPrefixes: addressPrefixes
+    }
+    dhcpOptions: {
+      dnsServers: []
+    }
+    // We define the subnets inline to avoid https://github.com/Azure/bicep/issues/4653
+    subnets: [
+      {
+        name: appServicePlanEgressSubnetName
+        properties: {
+          addressPrefix: appServicePlanEgressPrefixCidr
+          serviceEndpoints: [
+            {
+              service: 'Microsoft.AzureCosmosDB'
+              locations: [
+                '*'
+              ]
+            }
+            {
+              service: 'Microsoft.Storage'
+              locations: storageLocations
+            }
+          ]
+          delegations: [
+            {
+              name: '0'
+              properties: {
+                serviceName: 'Microsoft.Web/serverFarms'
+              }
+              type: 'Microsoft.Network/virtualNetworks/subnets/delegations'
+            }
+          ]
+          privateEndpointNetworkPolicies: 'Enabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+        type: 'Microsoft.Network/virtualNetworks/subnets'
+      }
+      {
+        name: privateEndpointsSubnetName
+        properties: {
+          addressPrefix: privateEndpointsCidr
+          networkSecurityGroup: {
+            id: networkSecurityGroupId
+          }
+          serviceEndpoints: [
+            {
+              service: 'Microsoft.AzureCosmosDB'
+              locations: [
+                '*'
+              ]
+            }
+            {
+              service: 'Microsoft.Storage'
+              locations: storageLocations
+            }
+          ]
+          delegations: []
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+        type: 'Microsoft.Network/virtualNetworks/subnets'
+      }
+      // TODO:DTFS-6422 Work out what these demo resources are for.
+      // Gabriel says it is a very old legacy item.
```

Comment:
yup, it needs to go. there was a 4th environment called `demo` up until 2022. That got decommissioned. We don't need to replicate it now so we should remove anything related to it